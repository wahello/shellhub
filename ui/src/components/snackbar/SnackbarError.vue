<template>
  <fragment>
    <v-snackbar
      v-model="snackbar"
      :timeout="4000"
      color="#bd4147"
      top
      outlined
      text
    >
      {{ message }}
    </v-snackbar>
  </fragment>
</template>

<script>

export default {
  name: 'SnackbarError',

  props: {
    typeMessage: {
      type: String,
      required: true,
    },

    mainContent: {
      type: String,
      default: '',
      required: false,
    },
  },

  computed: {
    snackbar: {
      get() {
        return this.$store.getters['snackbar/snackbarError'];
      },

      set() {
        this.$store.dispatch('snackbar/unsetShowStatusSnackbarError');
      },
    },

    message() {
      switch (this.typeMessage) {
      case 'loading':
        return `Loading the ${this.mainContent} has failed, please try again.`;
      case 'action':
        return `The ${this.mainContent} request has failed, please try again.`;
      case 'association':
        return 'There is no namespace associated with your account.';
      case 'notRequest':
        return `The ${this.mainContent} has failed, please try again.`;
      case 'incorrect':
        return `Incorrect ${this.mainContent} information, please try again.`;
      case 'deviceChoice':
        return 'You need to select 3 devices.';
      default:
        return 'The request has failed, please try again.';
      }
    },
  },
};

</script>
