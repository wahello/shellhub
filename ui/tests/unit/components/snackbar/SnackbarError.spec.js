import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import SnackbarError from '@/components/snackbar/SnackbarError';

describe('SnackbarError', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const snackbarError = true;
  let typeMessage = '';
  let mainContent = '';
  let message = '';

  const store = new Vuex.Store({
    namespaced: true,
    state: {
      snackbarError,
    },
    getters: {
      'snackbar/snackbarError': (state) => state.snackbarError,
    },
    actions: {
      'snackbar/unsetShowStatusSnackbarError': () => {},
    },
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to loading.
  ///////

  describe('Loading message type', () => {
    typeMessage = 'loading';
    mainContent = 'dashboard';
    message = `Loading the ${mainContent} has failed, please try again.`;

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to association.
  ///////

  describe('Association message type', () => {
    typeMessage = 'association';
    message = 'There is no namespace associated with your account.';

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to not request.
  ///////

  describe('NotRequest message type', () => {
    typeMessage = 'notRequest';
    mainContent = 'deviceDelete';
    message = `The ${mainContent} has failed, please try again.`;

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to incorrect.
  ///////

  describe('Incorrect message type', () => {
    typeMessage = 'incorrect';
    mainContent = 'deviceDelete';
    message = `Incorrect ${mainContent} information, please try again.`;

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to choice
  // devices.
  ///////

  describe('Incorrect message type', () => {
    typeMessage = 'deviceChoice';
    mainContent = '';
    message = 'You need to select 3 devices.';

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });

  ///////
  // In this case, the main objective is to change the message.
  // For this test to work, the message type is changed to default.
  ///////

  describe('Default message type', () => {
    typeMessage = 'default';
    message = 'The request has failed, please try again.';

    beforeEach(() => {
      wrapper = shallowMount(SnackbarError, {
        store,
        localVue,
        stubs: ['fragment'],
        propsData: { typeMessage, mainContent },
      });
    });

    ///////
    // Component Rendering
    //////

    it('Is a Vue instance', () => {
      expect(wrapper).toBeTruthy();
    });
    it('Renders the component', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data and Props checking
    //////

    it('Process data in the computed', () => {
      expect(wrapper.vm.snackbar).toEqual(snackbarError);
      expect(wrapper.vm.message).toEqual(message);
    });
    it('Process data in the computed', () => {
      expect(wrapper.vm.message).toEqual(message);
    });
  });
});
