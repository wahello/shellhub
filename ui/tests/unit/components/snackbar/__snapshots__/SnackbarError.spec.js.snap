// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackbarError Association message type Renders the component 1`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;

exports[`SnackbarError Default message type Renders the component 1`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;

exports[`SnackbarError Incorrect message type Renders the component 1`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;

exports[`SnackbarError Incorrect message type Renders the component 2`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;

exports[`SnackbarError Loading message type Renders the component 1`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;

exports[`SnackbarError NotRequest message type Renders the component 1`] = `
<fragment-stub>
  <v-snackbar-stub color="#bd4147" outlined="true" tag="div" value="true" top="true" contentclass="" text="true" timeout="4000" transition="v-snack-transition">
    The request has failed, please try again.
  </v-snackbar-stub>
</fragment-stub>
`;
